app/src/main/java/com/github/epfl/meili/auth/SignInActivity.kt

```diff
@@ -6,17 +6,20 @@ import android.view.View
 import android.widget.Button
 import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
+import com.facebook.AccessToken
 import com.github.epfl.meili.R
 import com.github.epfl.meili.map.MapActivity
 import com.github.epfl.meili.util.UserPreferences
 
-class GoogleSignInActivity : AppCompatActivity() {
+class SignInActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_google_sign_in)
-
-        Auth.setAuthenticationService(FirebaseAuthenticationService())
+        if (AccessToken.getCurrentAccessToken() != null) {
+            Auth.setAuthenticationService(FacebookAuthenticationService())
+        } else
+            Auth.setAuthenticationService(FirebaseAuthenticationService())
```

Comment:
I think this is not required since it's already done in the Logo Activity which always comes before the sign in activity, what do u think?